platform/android/variant/callable_jni.cpp

```diff
@@ -0,0 +1,117 @@
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "callable_jni.h"
+
+#include "jni_utils.h"
+
+#include "core/error/error_macros.h"
+#include "core/object/object.h"
+
+static Callable _generate_callable(JNIEnv *p_env, jlong p_object_id, jstring p_method_name, jobjectArray p_parameters) {
+	Object *obj = ObjectDB::get_instance(ObjectID(p_object_id));
+	ERR_FAIL_NULL_V(obj, Callable());
+
+	String str_method = jstring_to_string(p_method_name, p_env);
+
+	int count = p_env->GetArrayLength(p_parameters);
+
+	Variant *args = (Variant *)alloca(sizeof(Variant) * count);
+	const Variant **argptrs = (const Variant **)alloca(sizeof(Variant *) * count);
+
+	for (int i = 0; i < count; i++) {
+		jobject jobj = p_env->GetObjectArrayElement(p_parameters, i);
+		ERR_FAIL_NULL_V(jobj, Callable());
+		memnew_placement(&args[i], Variant(_jobject_to_variant(p_env, jobj)));
+		argptrs[i] = &args[i];
+		p_env->DeleteLocalRef(jobj);
+	}
+
+	return Callable(obj, str_method).bindp(argptrs, count);
+}
+
+extern "C" {
+JNIEXPORT jobject JNICALL Java_org_godotengine_godot_variant_Callable_nativeCall(JNIEnv *p_env, jclass p_clazz, jlong p_native_callable, jobjectArray p_parameters) {
+	const Variant *callable_variant = reinterpret_cast<const Variant *>(p_native_callable);
+	ERR_FAIL_NULL_V(callable_variant, nullptr);
+	if (callable_variant->get_type() != Variant::CALLABLE) {
+		return nullptr;
+	}
+
+	int count = p_env->GetArrayLength(p_parameters);
+
+	Variant *args = (Variant *)alloca(sizeof(Variant) * count);
+	const Variant **argptrs = (const Variant **)alloca(sizeof(Variant *) * count);
+
+	for (int i = 0; i < count; i++) {
+		jobject jobj = p_env->GetObjectArrayElement(p_parameters, i);
+		ERR_FAIL_NULL_V(jobj, nullptr);
+		memnew_placement(&args[i], Variant(_jobject_to_variant(p_env, jobj)));
+		argptrs[i] = &args[i];
+		p_env->DeleteLocalRef(jobj);
+	}
+
+	Callable callable = *callable_variant;
+	if (callable.is_valid()) {
+		Callable::CallError err;
+		Variant result;
+		callable.callp(argptrs, count, result, err);
+		jvalret jresult = _variant_to_jvalue(p_env, result.get_type(), &result, true);
+		return jresult.obj;
+	} else {
+		return nullptr;
+	}
```

Comment:
I think this has the same problem...

Reply:
It looks like `Java_org_godotengine_godot_plugin_GodotPlugin_nativeEmitSignal()` (which isn't modified in this PR) also shares this same code